rdrf/registry/groups/registration/parent_with_patient/common.py

```diff
@@ -124,31 +109,54 @@ def process(self):
         process_notification(registry_code, EventType.NEW_PATIENT, template_data)
         logger.debug("Registration process - sent notification for NEW_PATIENT")
 
-    def _create_parent(self, request):
+    def get_registration_workflow(self):
+        return PatientWithParentRegistrationWorkflow(None, None)
+
+    def _create_parent(self):
+        form_data = self.form.cleaned_data
         parent_guardian = ParentGuardian.objects.create(
-            first_name=request.POST["parent_guardian_first_name"],
-            last_name=request.POST["parent_guardian_last_name"],
-            date_of_birth=request.POST["parent_guardian_date_of_birth"],
-            gender=request.POST["parent_guardian_gender"],
-            address=request.POST["parent_guardian_address"],
-            suburb=request.POST["parent_guardian_suburb"],
-            state=request.POST["parent_guardian_state"],
-            postcode=request.POST["parent_guardian_postcode"],
-            country=request.POST["parent_guardian_country"],
-            phone=request.POST["parent_guardian_phone"],
+            first_name=form_data["parent_guardian_first_name"],
+            last_name=form_data["parent_guardian_last_name"],
+            date_of_birth=form_data["parent_guardian_date_of_birth"],
+            gender=form_data["parent_guardian_gender"],
+            address=form_data["parent_guardian_address"],
+            suburb=form_data["parent_guardian_suburb"],
+            state=form_data["parent_guardian_state"],
+            postcode=form_data["parent_guardian_postcode"],
+            country=form_data["parent_guardian_country"],
+            phone=form_data["parent_guardian_phone"],
         )
         return parent_guardian
 
-    def _create_patient_address(self, patient, request, address_type="Postal"):
-        same_address = "same_address" in request.POST
-
-        address = PatientAddress.objects.create(
-            patient=patient,
-            address_type=self.get_address_type(address_type),
-            address=request.POST["parent_guardian_address"] if same_address else request.POST["address"],
-            suburb=request.POST["parent_guardian_suburb"] if same_address else request.POST["suburb"],
-            state=request.POST["parent_guardian_state"] if same_address else request.POST["state"],
-            postcode=request.POST["parent_guardian_postcode"] if same_address else request.POST["postcode"],
-            country=request.POST["parent_guardian_country"] if same_address else request.POST["country"]
-        )
-        return address
+    def _create_django_user(self, django_user, registry, is_parent, groups=[], is_clinician=False):
```

Comment:
This and the correspondent method in the ParentRegistration look suspicious to me.

The name of the method is misleading as it doesn't create the django user. Should be `setup_django_user`?
The mixture of groups and is_parent, is_clinician parameters, when is_parent and is_clinician are done through groups as well seems ugly. Is the groups parameter used at all?

Can we simplify by having a _setup_django_user method instead of this in the base class and maybe do the setting of groups in another method that just takes the group name as an argument?

Reply:
Updated.